src/mitol/authentication/views/djoser_views.py

```diff
@@ -0,0 +1,77 @@
+"""
+Custom views to override default djoser behaviours
+"""
+from django.conf import settings
+from django.contrib.auth import update_session_auth_hash
+from django.core import mail as django_mail
+from djoser.utils import ActionViewMixin
+from djoser.views import UserViewSet
+from djoser.email import PasswordResetEmail as DjoserPasswordResetEmail
+from rest_framework import status
+from rest_framework.decorators import action
+from anymail.message import AnymailMessage
+from mitol.mail.api import render_email_templates, send_message
+
+
+class CustomDjoserAPIView(UserViewSet, ActionViewMixin):
+    """
+    Overrides the post method of a Djoser view to update session after successful password change
+    """
+
+
+class CustomDjoserAPIView(UserViewSet, ActionViewMixin):
+    """
+    Overrides the post method of a Djoser view to update session after successful password change
+    """
+
+    @action(["post"], detail=False)
+    def set_password(self, request, *args, **kwargs):
+        """
+        Overrides CustomDjoserAPIView.set_password to update the session after a successful
+        password change. Without this explicit refresh, the user's session would be
+        invalid and they would be logged out.
+        """
+        response = super().set_password(request, *args, **kwargs)
+        if response.status_code in (status.HTTP_200_OK, status.HTTP_204_NO_CONTENT):
+            update_session_auth_hash(self.request, self.request.user)
+        return response
+
+class CustomPasswordResetEmail(DjoserPasswordResetEmail):
+    """Custom class to modify base functionality in Djoser's PasswordResetEmail class"""
+
+    
+    def get_render_email_templates(slef, template_name, context):
+        """Funtion to make render email app project independent"""
+        return render_email_templates(template_name, context)
+
+    def send_message(template_message):
+        send_message(template_message)    
```

Comment:
There are typos/bugs here but I think you can just remove these in favor of calling the functions directly

Reply:
In case of xPRO `send_messages` just logs the messages, However MITxOnline logs and and raises an EmailSendFailure EmailSendFailureException messages which errored out.